add metadata yaml generation

seemk opened the pull request:
If possible the instrumentation target library versions are taken from dependency READMEs (however in the future a better mechanism is needed to gather these automatically, something to discuss in the JS sig).

Example yaml can be seen at https://github.com/signalfx/splunk-otel-js/actions/runs/7474145786/artifacts/1159275535

theletterf reviewed (changes requested):
Great! There are a few things that need addressing:

- There seem to be a lack of indentation for certain items. For example, items inside each first level object are not indented. Not sure if this would pose a problem to YAML parsers, but I'd much rather have indentation for readability and browsing purposes. 
The same indentation issues appears in nested items, such as the dependencies within the resource detectors.

- There is no first level `instrumentations:` object — keys start right after the settings.

- No resource attribute information nor dependencies info, but I think it's OK for the latter to just resort to npm, especially if there are MANY dependencies. WDYT?

- To preserve the same structure used by .NET, please rename the name key in settings to env.

With this, I think we'd already have an useful artifact for testing.

seemk commented:
> Great! There are a few things that need addressing:
> 
> * There seem to be a lack of indentation for certain items. For example, items inside each first level object are not indented. Not sure if this would pose a problem to YAML parsers, but I'd much rather have indentation for readability and browsing purposes.
>   The same indentation issues appears in nested items, such as the dependencies within the resource detectors.

Should have better indenting now and verified it's parseable.

> * There is no first level `instrumentations:` object — keys start right after the settings.

Fixed

> * No resource attribute information nor dependencies info, but I think it's OK for the latter to just resort to npm, especially if there are MANY dependencies. WDYT?

Added resource detectors, I can add the dependency list for sure, but is there an actual need for it? Should they be first level dependencies?

> * To preserve the same structure used by .NET, please rename the name key in settings to env.
> 

Done!

theletterf commented:
@seemk Could we try with adding them as first-level deps and see what comes out? If it's not too much work, that is.

seemk commented:
> @seemk Could we try with adding them as first-level deps and see what comes out? If it's not too much work, that is.

Added first level dependencies, the output is now

```
component: Splunk Distribution of OpenTelemetry JavaScript
version: 2.6.1
settings:
- env: OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT
  description: Maximum allowed attribute value size
  default: 12000
  type: number
  category: instrumentation
- env: OTEL_BSP_SCHEDULE_DELAY
  description: The delay in milliseconds between 2 consecutive bath span processor exports.
  default: 500
  type: number
  category: instrumentation
- env: OTEL_EXPORTER_OTLP_CERTIFICATE
  description: Path to a certificate to use when verifying a server's TLS credentials.
  default: 
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE
  description: Path to a certificate to use when verifying a client's TLS credentials.
  default: 
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_CLIENT_KEY
  description: Path to client's private key to use in mTLS communication in PEM format.
  default: 
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_ENDPOINT
  description: The OTLP endpoint to export to.
  default: http://localhost:4317
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_TRACES_PROTOCOL
  description: Chooses the trace exporter protocol. Allowed values are grpc and http/protobuf
  default: grpc
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_METRICS_PROTOCOL
  description: Chooses the metric exporter protocol. Allowed values are grpc and http/protobuf
  default: grpc
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_PROTOCOL
  description: The protocol to use for OTLP exports.
  default: grpc
  type: string
  category: exporter
- env: OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
  description: The traces OTLP endpoint to export to.
  default: http://localhost:4317
  type: string
  category: exporter
- env: OTEL_INSTRUMENTATION_COMMON_DEFAULT_ENABLED
  description: Whether to activate all the embedded instrumentations. When you set this setting to false, use OTEL_INSTRUMENTATION_<NAME>_ENABLED=true to selectively turn on instrumentations.
  default: true
  type: boolean
  category: instrumentation
- env: OTEL_LOG_LEVEL
  description: Log level for the OpenTelemetry diagnostic console logger. To activate debug logging, set the debug value. Available values are error, info, debug, and verbose.
  default: none
  type: string
  category: general
- env: OTEL_METRIC_EXPORT_INTERVAL
  description: The interval, in milliseconds, of metrics collection and exporting.
  default: 30000
  type: number
  category: exporter
- env: OTEL_METRICS_EXPORTER
  description: Comma-separated list of metrics exporter to use. To output to the console, set the variable to console. If set to none, metric exports are turned off.
  default: otlp
  type: string
  category: exporter
- env: OTEL_PROPAGATORS
  description: Comma-separated list of propagators you want to use.
  default: tracecontext,baggage
  type: string
  category: general
- env: OTEL_SERVICE_NAME
  description: Name of the service or application you’re instrumenting. Takes precedence over the service name defined in the OTEL_RESOURCE_ATTRIBUTES variable.
  default: unnamed-node-service
  type: string
  category: general
- env: OTEL_SPAN_LINK_COUNT_LIMIT
  description: Maximum number of links per span.
  default: 1000
  type: number
  category: general
- env: OTEL_TRACES_EXPORTER
  description: Comma-separated list of trace exporters to use. To output to the console, set the variable to console.
  default: otlp
  type: string
  category: exporter
- env: SPLUNK_ACCESS_TOKEN
  description: A Splunk authentication token that lets exporters send data directly to Splunk Observability Cloud. Required if you need to send data to the Splunk Observability Cloud ingest endpoint.
  default: 
  type: string
  category: general
- env: SPLUNK_INSTRUMENTATION_METRICS_ENABLED
  description: Emit metrics from instrumentation (e.g. http.server.duration)
  default: false
  type: boolean
  category: instrumentation
- env: SPLUNK_METRICS_ENABLED
  description: Activates metrics collection.
  default: false
  type: boolean
  category: general
- env: SPLUNK_METRICS_ENDPOINT
  description: The metrics endpoint. Takes precedence over OTEL_EXPORTER_OTLP_METRICS_ENDPOINT. When SPLUNK_REALM is used, the default value is https://ingest.<realm>.signalfx.com/v2/datapoint/otlp.
  default: 
  type: string
  category: general
- env: SPLUNK_PROFILER_CALL_STACK_INTERVAL
  description: Frequency with which call stacks are sampled, in milliseconds.
  default: 1000
  type: number
  category: profiler
- env: SPLUNK_PROFILER_ENABLED
  description: Activates AlwaysOn CPU profiling.
  default: false
  type: boolean
  category: profiler
- env: SPLUNK_PROFILER_LOGS_ENDPOINT
  description: The collector endpoint for profiler logs.
  default: http://localhost:4317
  type: string
  category: profiler
- env: SPLUNK_PROFILER_MEMORY_ENABLED
  description: Activates memory profiling for AlwaysOn Profiling.
  default: false
  type: string
  category: profiler
- env: SPLUNK_REALM
  description: The name of your organization’s realm, for example, us0. When you set the realm, telemetry is sent directly to the ingest endpoint of Splunk Observability Cloud, bypassing the Splunk Distribution of OpenTelemetry Collector.
  default: 
  type: string
  category: general
- env: SPLUNK_REDIS_INCLUDE_COMMAND_ARGS
  description: Whether to include the full Redis query in db.statement span attributes when using the Redis instrumentation.
  default: false
  type: boolean
  category: instrumentation
- env: SPLUNK_RUNTIME_METRICS_COLLECTION_INTERVAL
  description: The interval, in milliseconds, during which GC and event loop statistics are collected.
  default: 5000
  type: number
  category: instrumentation
- env: SPLUNK_RUNTIME_METRICS_ENABLED
  description: Activates the collection and export of runtime metrics. Runtime metrics are only sent if the SPLUNK_METRICS_ENABLED environment variable is set to true or if memory profiling is activated.
  default: true
  type: boolean
  category: instrumentation
- env: SPLUNK_TRACE_RESPONSE_HEADER_ENABLED
  description: Activates the addition of server trace information to HTTP response headers.
  default: true
  type: boolean
  category: general
- env: SPLUNK_TRACING_ENABLED
  description: Enables tracing.
  default: true
  type: boolean
  category: instrumentation
instrumentations:
  - keys:
    - "@opentelemetry/instrumentation-amqplib"
    instrumented_components:
      - name: "amqplib"
        supported_versions: ">=0.5.5"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-aws-sdk"
    instrumented_components:
      - name: "aws-sdk and @aws-sdk"
        supported_versions: "2.x,3.x"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-bunyan"
    instrumented_components:
      - name: "bunyan"
        supported_versions: "bunyan@^1.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-cassandra-driver"
    instrumented_components:
      - name: "cassandra-driver"
        supported_versions: ">=4.4 <5.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-connect"
    instrumented_components:
      - name: "connect"
        supported_versions: "^3.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-dataloader"
    instrumented_components:
      - name: "dataloader"
        supported_versions: "^2.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-dns"
    instrumented_components:
      - name: "dns"
        supported_versions: "*"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-express"
    instrumented_components:
      - name: "express"
        supported_versions: "^4.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-fastify"
    instrumented_components:
      - name: "fastify"
        supported_versions: "fastify: ^3.0.0 || ^4.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-generic-pool"
    instrumented_components:
      - name: "generic-pool"
        supported_versions: ">=2.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-graphql"
    instrumented_components:
      - name: "graphql"
        supported_versions: ">=14 <16"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-grpc"
    instrumented_components:
      - name: "@grpc/grpc-js"
        supported_versions: "1.x"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-hapi"
    instrumented_components:
      - name: "hapi"
        supported_versions: ">=17.0.0 <21"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-http"
    instrumented_components:
      - name: "http"
        supported_versions: "*"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-ioredis"
    instrumented_components:
      - name: "ioredis"
        supported_versions: ">=2.0.0 <6"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-knex"
    instrumented_components:
      - name: "knex"
        supported_versions: ">=0.10.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-koa"
    instrumented_components:
      - name: "koa"
        supported_versions: "koa: ^2.0.0,@koa/router: >=8"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-memcached"
    instrumented_components:
      - name: "memcached"
        supported_versions: ">=2.2"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-mongodb"
    instrumented_components:
      - name: "mongodb"
        supported_versions: ">=3.3 <7"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-mongoose"
    instrumented_components:
      - name: "mongoose"
        supported_versions: ">=5.9.7 <7"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-mysql"
    instrumented_components:
      - name: "mysql"
        supported_versions: "2.x"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-mysql2"
    instrumented_components:
      - name: "mysql2"
        supported_versions: ">= 1.4.2, < 4.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-nestjs-core"
    instrumented_components:
      - name: "@nestjs/core"
        supported_versions: ">=4.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-net"
    instrumented_components:
      - name: "net"
        supported_versions: "*"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-pg"
    instrumented_components:
      - name: "pg"
        supported_versions: ">=8 <9"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-pino"
    instrumented_components:
      - name: "pino"
        supported_versions: ">=5.14.0 <9"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-redis"
    instrumented_components:
      - name: "redis"
        supported_versions: "^2.6.0,3.x"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-redis-4"
    instrumented_components:
      - name: "redis"
        supported_versions: "4.x"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-restify"
    instrumented_components:
      - name: "restify"
        supported_versions: ">=4.0.0 <12"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-router"
    instrumented_components:
      - name: "router"
        supported_versions: ">=1.0.0"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-tedious"
    instrumented_components:
      - name: "tedious"
        supported_versions: ">=1.11.0 <=15"
    support: community
  - keys:
    - "@opentelemetry/instrumentation-winston"
    instrumented_components:
      - name: "winston"
        supported_versions: "1.x, 2.x, 3.x"
    support: community
  - keys:
    - "splunk-opentelemetry-instrumentation-elasticsearch"
    instrumented_components:
      - name: "@elastic/elasticsearch"
        supported_versions: ">=5 <8"
    support: supported
  - keys:
    - "splunk-opentelemetry-instrumentation-kafkajs"
    instrumented_components:
      - name: "kafkajs"
        supported_versions: "*"
    support: supported
  - keys:
    - "splunk-opentelemetry-instrumentation-sequelize"
    instrumented_components:
      - name: "sequelize"
        supported_versions: "*"
    support: supported
  - keys:
    - "splunk-opentelemetry-instrumentation-typeorm"
    instrumented_components:
      - name: "typeorm"
        supported_versions: ">0.2.28"
    support: supported
resource_detectors:
  - key: PROCESS
    description: Process info detector
    attributes:
      - id: process.pid
      - id: process.executable.path
      - id: process.runtime.version
      - id: process.runtime.name
    support: supported
  - key: OS
    description: Operating system detector
    attributes:
      - id: os.type
      - id: os.description
    support: supported
  - key: HOST
    description: Host detector
    attributes:
      - id: host.name
      - id: host.arch
    support: supported
  - key: CONTAINER
    description: Container ID detector
    attributes:
      - id: container.id
    support: supported
  - key: DISTRO
    description: Distribution version detector
    attributes:
      - id: splunk.distro.version
    support: supported
dependencies:
  - name: "is-promise"
    version: "^4.0.0"
    stability: stable
  - name: "nan"
    version: "^2.18.0"
    stability: stable
  - name: "node-gyp-build"
    version: "^4.7.1"
    stability: stable
    source_href: "https://github.com/prebuild/node-gyp-build"
  - name: "protobufjs"
    version: "^7.2.5"
    stability: stable
    source_href: "https://protobufjs.github.io/protobuf.js/"
  - name: "semver"
    version: "^7.5.4"
    stability: stable
  - name: "@grpc/grpc-js"
    version: "^1.8.19"
    stability: stable
    source_href: "https://grpc.io/"
  - name: "@grpc/proto-loader"
    version: "^0.7.8"
    stability: stable
    source_href: "https://grpc.io/"
  - name: "@opentelemetry/api"
    version: "^1.3.0"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/api"
  - name: "@opentelemetry/context-async-hooks"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-context-async-hooks"
  - name: "@opentelemetry/core"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-core"
  - name: "@opentelemetry/exporter-metrics-otlp-grpc"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-exporter-metrics-otlp-grpc"
  - name: "@opentelemetry/exporter-metrics-otlp-proto"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-exporter-metrics-otlp-proto"
  - name: "@opentelemetry/exporter-trace-otlp-grpc"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/exporter-trace-otlp-grpc"
  - name: "@opentelemetry/exporter-trace-otlp-proto"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/exporter-trace-otlp-proto"
  - name: "@opentelemetry/instrumentation"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-instrumentation"
  - name: "@opentelemetry/instrumentation-amqplib"
    version: "0.33.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-amqplib#readme"
  - name: "@opentelemetry/instrumentation-aws-sdk"
    version: "0.37.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#readme"
  - name: "@opentelemetry/instrumentation-bunyan"
    version: "0.34.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-bunyan#readme"
  - name: "@opentelemetry/instrumentation-cassandra-driver"
    version: "0.34.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-cassandra#readme"
  - name: "@opentelemetry/instrumentation-connect"
    version: "0.32.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-connect#readme"
  - name: "@opentelemetry/instrumentation-dataloader"
    version: "0.5.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-dataloader#readme"
  - name: "@opentelemetry/instrumentation-dns"
    version: "0.32.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-dns#readme"
  - name: "@opentelemetry/instrumentation-express"
    version: "0.34.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-express#readme"
  - name: "@opentelemetry/instrumentation-fastify"
    version: "0.32.5"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-fastify#readme"
  - name: "@opentelemetry/instrumentation-generic-pool"
    version: "0.32.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-generic-pool#readme"
  - name: "@opentelemetry/instrumentation-graphql"
    version: "0.36.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-graphql#readme"
  - name: "@opentelemetry/instrumentation-grpc"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-instrumentation-grpc"
  - name: "@opentelemetry/instrumentation-hapi"
    version: "0.33.2"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-hapi#readme"
  - name: "@opentelemetry/instrumentation-http"
    version: "0.45.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-instrumentation-http"
  - name: "@opentelemetry/instrumentation-ioredis"
    version: "0.36.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-ioredis#readme"
  - name: "@opentelemetry/instrumentation-knex"
    version: "0.32.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-knex#readme"
  - name: "@opentelemetry/instrumentation-koa"
    version: "0.36.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-koa#readme"
  - name: "@opentelemetry/instrumentation-memcached"
    version: "0.32.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-memcached#readme"
  - name: "@opentelemetry/instrumentation-mongodb"
    version: "0.38.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-mongodb#readme"
  - name: "@opentelemetry/instrumentation-mongoose"
    version: "0.33.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-mongoose#readme"
  - name: "@opentelemetry/instrumentation-mysql"
    version: "0.34.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-mysql#readme"
  - name: "@opentelemetry/instrumentation-mysql2"
    version: "0.34.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-mysql2#readme"
  - name: "@opentelemetry/instrumentation-nestjs-core"
    version: "0.33.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-nestjs-core#readme"
  - name: "@opentelemetry/instrumentation-net"
    version: "0.32.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-net#readme"
  - name: "@opentelemetry/instrumentation-pg"
    version: "0.37.1"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-pg#readme"
  - name: "@opentelemetry/instrumentation-pino"
    version: "0.34.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-pino#readme"
  - name: "@opentelemetry/instrumentation-redis"
    version: "0.35.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-redis#readme"
  - name: "@opentelemetry/instrumentation-redis-4"
    version: "0.35.5"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-redis-4#readme"
  - name: "@opentelemetry/instrumentation-restify"
    version: "0.34.2"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-restify#readme"
  - name: "@opentelemetry/instrumentation-router"
    version: "0.33.3"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-router#readme"
  - name: "@opentelemetry/instrumentation-tedious"
    version: "0.6.4"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-tedious#readme"
  - name: "@opentelemetry/instrumentation-winston"
    version: "0.33.0"
    stability: experimental
    source_href: "https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-winston#readme"
  - name: "@opentelemetry/propagator-b3"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-propagator-b3"
  - name: "@opentelemetry/resources"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-resources"
  - name: "@opentelemetry/sdk-metrics"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/sdk-metrics"
  - name: "@opentelemetry/sdk-trace-base"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-sdk-trace-base"
  - name: "@opentelemetry/sdk-trace-node"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-sdk-trace-node"
  - name: "@opentelemetry/semantic-conventions"
    version: "1.18.1"
    stability: stable
    source_href: "https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-semantic-conventions"
```

theletterf commented:
Seems reasonable! Let's go with it.

seemk commented:
> Seems reasonable! Let's go with it.

@theletterf You need to approve as well in this case
